### PR TITLE
PR: Add branding for Kite completion items

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

b

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/completion/kite/providers/document.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
     'unknown': CompletionItemKind.TEXT
 }
 
+KITE_COMPLETION = 'Kite'
```
Comment (its-dhung) [inline spyder/plugins/completion/kite/providers/document.py]: I would change this to:
```python
KITE_COMPLETION = '[kite]'
```

**Comment by ccordoba12** [comment]
@dalthviz, I don't like this approach too much. I'd prefer to use Kite's icon instead of our current icons to signal that a completion is coming from Kite.

The problem is this approach is a bit misleading because a regular class (like `ImportError`, in your screenshot above) seems to be become a special class (called `Kite class`) when the user select the Kite entry for it.

Thoughts @its-dhung, @goanpeca?

**Comment by goanpeca** [comment]
is there a reason why the message of the type of completion kite or not appears only when selecting it, instead o seeing, for example, the icon all the time?

Also we could considering showing this icon to the right of the box (instead of before the `class` text)

**Comment by dalthviz** [comment]
@goanpeca I think the initial idea was to show the provider of the completion only for the current selected item, but maybe is better to show the provider (probably the icon) always?

**Comment by goanpeca** [comment]
>but maybe is better to show the provider (probably the icon) always?

It would make it clear, besides we could explain on the initial part, that the completion widgets obtined by kite will display the icon (so users have a clear expectation)

**Comment by metalogical** [comment]
I'm generally in agreement with all the comments here. The approximate behavior we use for other editors is:
- if we're showing the icon, we show it for all completions (we overlay it on the left, but the right is fine as well).
- if we're showing the `[kite]` text, then we show it only for the selected completion to reduce visual noise.

**Comment by its-dhung** [comment]
To reitreate what @metalogical said, if we use Kite's icon, I would place it on the left. However, if we have to place it on the right, I would put it _after_ the type of completion e.g. `class [icon]`

**Comment by dalthviz** [comment]
@its-dhung is the new preview what you spect (I updated the preview gif in the initial comment of the PR)?

**Comment by its-dhung** [comment]
@dalthviz yes that looks great!

File: spyder/plugins/editor/widgets/completion.py [hunk 2]
```diff
@@ -253,14 +257,42 @@ def update_list(self, filter_text):
...
+        item_provider = item_info['provider']
+        item_type = self.ICONS_MAP.get(item_info['kind'], 'no_match')
+        item_label = item_info['label']
+        icon_provider = ("iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0l"
+                         "EQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=")
+        img_height = height - 2
+        img_width = img_height * 0.8
+
+        if item_provider == KITE_COMPLETION:
+            icon_provider = ima.get_kite_icon()
```
Comment (its-dhung) [inline spyder/plugins/editor/widgets/completion.py]: The icon appears to be blurry due to the resizing being done by the GUI.

image

You can fix this by resizing the image here:

```python
kite_height = img_height
kite_width = (416.14/526.8) * kite_height
icon_provider = ima.get_kite_icon()
icon_provider = ima.base64_from_icon_obj(
    icon_provider, kite_width, kite_height)
```

This is what it looks like when manually resized:

image

**Comment by goanpeca** [comment]
@dalthviz is this ready?

**Comment by dalthviz** [comment]
Yep @goanpeca I think so 👍

**Comment by goanpeca** [comment]
The windows error on azure is not related to the PR. Merging

**Comment by sergeyFlakon** [comment]
I see branding displayed correctly on Commit: (13c208033)

**Pull request merged**